test/examples/nodejs_postgresql_test.go

```diff
@@ -0,0 +1,539 @@
+package examples_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+	"github.com/redhat-developer/service-binding-operator/test/examples/util"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	exampleName = "nodejs_postgresql"
+	operatorsNS = "openshift-operators"
+	oprName     = "service-binding-operator"
+	appNS       = "service-binding-demo"
+
+	nodeJsApp = "https://github.com/pmacik/nodejs-rest-http-crud"
+	appName   = "nodejs-rest-http-crud"
+
+	clusterAvailable = false
+
+	pkgManifest                      = "db-operators"
+	bckSvc                           = "postgresql-operator"
+	dbName                           = "db-demo"
+	sbr                              = ""
+	podNameSBO, podNameBackingSvcOpr string
+)
+
+func TestNodeJSPostgreSQL(t *testing.T) {
+
+	t.Run("set-example-dir", SetExampleDir)
+	t.Run("get-oc-status", GetOCStatus)
+
+	t.Run("install-service-binding-operator", MakeInstallServiceBindingOperator)
+	t.Run("install-backing-service-operator", MakeInstallBackingServiceOperator)
+	t.Run("create-project", CreateProject)
+	t.Run("import-nodejs-app", ImportNodeJSApp)
+	t.Run("create-backing-db-instance", CreateBackingDbInstance)
+	t.Run("createservice-binding-request", CreateServiceBindingRequest)
+	t.Run("delete-project", DeleteProject)
+	t.Run("uninstall-backing-db-instance", UninstallBackingServiceOperator)
+	t.Run("uninstall-service-binding-operator", UninstallServiceBindingOperator)
+}
+
+//SetExampleDir tests that the current example directory was set as a working directory for running the commands.
+func SetExampleDir(t *testing.T) {
```

Comment:
Why functions `GetExampleDir` and `SetExampleDir` are in different packages?

I can see `SetDir` is in `util` package ->
https://github.com/redhat-developer/service-binding-operator/pull/441/files#diff-df905039daccb0b0c7e51d4480f3adf1R50

Reply:
If `SetExampleDir` is a test function with `t *testing.T`.
Shouldn't the name of the function start with `Test<something>`?

Reply:
Assertion is done on SetExampleDir from with t.Run and can be used for other examples as well. hence placed GetExampleDir and SetDir is placed in util package

Reply:
if you notice TestNodeJSPostgreSQL() is a test function that runs number of tests for convenience. i noticed GetOCStatus() is the one i need to assert.

Reply:
> Shouldn't the name of the function start with `Test<something>`?

@DhritiShikhar `SetExampleDir` is a sub-test executed by the `t.Run` as a step of the e2e scenario. This approach is used for convenience in a development of the scenario. By using t.Run as in

Reply:
The reason we use the `t.Run` approach is that it makes it possible to make the test fail-fast when using the `require` package.